accepted/future-releases/constructor-tearoffs/feature-specification.md

```diff
@@ -295,32 +330,41 @@ Example:
 ```dart
 T id<T>(T value) => value;
-var idInt = id<int>; // Explicitly instantiated tear-off, saves on writing function types.
+var idInt = id<int>; // Explicitly instantiated tear-off, saves on function types.
 // and
 Type intList = List<int>; // In-line instantiated type literal.
 ```
 
-These grammar changes allows *type parameters* without following parenthesized arguments in places where we previously did not allow them. For example, this means that `<typeArguments>` becomes a *selector* by itself, not just followed by arguments.
+These grammar changes allow *type parameters* without following parenthesized arguments in places where we previously did not allow that. This means that `<typeArguments>` becomes a *selector* by itself, not just when followed by arguments.
+
+Explicit instantiation applies to any function value, whether tear-offs of instance methods or local, static and top-level function declarations, or first-class function values.
 
-It applies to instance methods as well as local, static and top-level function declarations. For instance methods, it applies to references of the form
+For an expression of the form <code>*e*\<*typeArgs*></code>, which is not followed by an argument list (that would turn it into a generic function invocation), the meaning of <code>*e*\<typeArgs></code> depends on the expression *e*:
 
-`instanceMethod<int>` (with implicit `this`), `object.instanceMethod<int>` (including `this`) and `super.instanceMethod<int>`.
+* If *e* denotes a generic class, mixin or type alias declaration (which means that *e* is an identifier, possibly a qualified identifier, which resolves to the class, mixin or type alias declaration), then <code>*e*\<*typeArgs*></code> is a type literal. _If followed by <code>.*id*</code> then that *id* must denote a constructor, which can then be either torn off or invoked. If followed by `==` or `!=` or any "stop-token", the expression evaluates to a `Type` object._
+* If *e* denotes a generic top-level, static or local function declaration (again *e* is an identifier or qualified identifier), that declaration must be a generic function declaration, and then <code>*e*\<*typeArgs*></code> performs an explicitly instantiated function tear-off, which works just like the current implicitly instantiated function tear-off except that the types are provided instead of inferred.
+* If *e* denotes a generic instance method (*e* has the form <code>*r*.*name*</code> and *r* has a static type for which *name* is a generic interface method), then <code>*e*\<*typeArgs*></code> performs an explicitly instantiated method tear-off, which works just like the current implicitly instantiated method tear-off except that the types are provided instead of inferred.
+* If *e* has a static type which is a generic callable object type (a non-function type with a generic method named `call`), then <code>*e*\<*typeArgs*></code> is equivalent to the instantiated method-tear off <code>*e*\.call<*typeArgs*></code>.
```

Comment:
OK, so we're basically accepting #1827 and doing it here.

Reply:
Yes.